packages/core/src/tracing/idlespan.ts

```diff
@@ -0,0 +1,318 @@
+/* eslint-disable max-lines */
+import type { TransactionContext } from '@sentry/types';
+import { logger, timestampInSeconds } from '@sentry/utils';
+
+import { DEBUG_BUILD } from '../debug-build';
+import { getClient } from '../exports';
+import type { Span } from './span';
+import { startSpanManual } from './trace';
+
+export const TRACING_DEFAULTS = {
+  idleTimeout: 1000,
+  finalTimeout: 30000,
+  heartbeatInterval: 5000,
+};
+
+const FINISH_REASON_TAG = 'finishReason';
+
+const FINISH_REASON_HEARTBEAT_FAILED = 'heartbeatFailed';
+const FINISH_REASON_IDLE_TIMEOUT = 'idleTimeout';
+const FINISH_REASON_FINAL_TIMEOUT = 'finalTimeout';
+const FINISH_REASON_EXTERNAL_FINISH = 'externalFinish';
+const FINISH_REASON_CANCELLED = 'cancelled';
+
+// unused
+const FINISH_REASON_DOCUMENT_HIDDEN = 'documentHidden';
+
+// unusued in this file, but used in BrowserTracing
+const FINISH_REASON_INTERRUPTED = 'interactionInterrupted';
+
+type IdleSpanFinishReason =
+  | typeof FINISH_REASON_CANCELLED
+  | typeof FINISH_REASON_DOCUMENT_HIDDEN
+  | typeof FINISH_REASON_EXTERNAL_FINISH
+  | typeof FINISH_REASON_FINAL_TIMEOUT
+  | typeof FINISH_REASON_HEARTBEAT_FAILED
+  | typeof FINISH_REASON_IDLE_TIMEOUT
+  | typeof FINISH_REASON_INTERRUPTED;
+
+interface IdleSpanOptions {
+  transactionContext: TransactionContext;
+  idleTimeout: number;
+  finalTimeout: number;
+  // TODO FN: Do we need this??
+  // customSamplingContext?: CustomSamplingContext;
```

Comment:
This is a general thing that already came up - do we need to add this to `startSpan` somehow? 🤔